pkg/storage/driver/secrets.go

```diff
@@ -0,0 +1,262 @@
+func (secrets *Secrets) Create(key string, rls *rspb.Release) error {
+	// set labels for secrets object meta data
+	var lbs labels
+
+	lbs.init()
+	lbs.set("CREATED_AT", strconv.Itoa(int(time.Now().Unix())))
+
+	// create a new secret to hold the release
+	obj, err := newSecretsObject(key, rls, lbs)
+	if err != nil {
+		secrets.Log("create: failed to encode release %q: %s", rls.Name, err)
+		return err
+	}
+	// push the secret object out into the kubiverse
+	if _, err := secrets.impl.Create(obj); err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			return ErrReleaseExists(rls.Name)
+		}
+
+		secrets.Log("create: failed to create: %s", err)
+		return err
+	}
+	return nil
+}
+
+// Update updates the Secret holding the release. If not found
+// the Secret is created to hold the release.
+func (secrets *Secrets) Update(key string, rls *rspb.Release) error {
+	// set labels for secrets object meta data
+	var lbs labels
+
+	lbs.init()
+	lbs.set("MODIFIED_AT", strconv.Itoa(int(time.Now().Unix())))
+
+	// create a new secret object to hold the release
+	obj, err := newSecretsObject(key, rls, lbs)
+	if err != nil {
+		secrets.Log("update: failed to encode release %q: %s", rls.Name, err)
+		return err
+	}
+	// push the secret object out into the kubiverse
+	_, err = secrets.impl.Update(obj)
+	if err != nil {
+		secrets.Log("update: failed to update: %s", err)
+		return err
+	}
+	return nil
+}
+
+// Delete deletes the Secret holding the release named by key.
+func (secrets *Secrets) Delete(key string) (rls *rspb.Release, err error) {
+	// fetch the release to check existence
+	if rls, err = secrets.Get(key); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, ErrReleaseExists(rls.Name)
+		}
+
+		secrets.Log("delete: failed to get release %q: %s", key, err)
+		return nil, err
+	}
+	// delete the release
+	if err = secrets.impl.Delete(key, &metav1.DeleteOptions{}); err != nil {
+		return rls, err
+	}
+	return rls, nil
+}
+
+// newSecretsObject constructs a kubernetes Secret object
+// to store a release. Each secret data entry is the base64
+// encoded string of a release's binary protobuf encoding.
+//
+// The following labels are used within each secret:
+//
+//    "MODIFIED_AT"    - timestamp indicating when this secret was last modified. (set in Update)
+//    "CREATED_AT"     - timestamp indicating when this secret was created. (set in Create)
+//    "VERSION"        - version of the release.
+//    "STATUS"         - status of the release (see proto/hapi/release.status.pb.go for variants)
+//    "OWNER"          - owner of the secret, currently "TILLER".
+//    "NAME"           - name of the release.
+//
+func newSecretsObject(key string, rls *rspb.Release, lbs labels) (*api.Secret, error) {
+	const owner = "TILLER"
+
+	// encode the release
+	s, err := encodeRelease(rls)
+	if err != nil {
+		return nil, err
+	}
+
+	if lbs == nil {
+		lbs.init()
+	}
+
+	// apply labels
+	lbs.set("NAME", rls.Name)
```

Comment:
Because these are used in Secrets and ConfigMaps, it may be useful to define them as constants